### DISCUSSION TITLE
最新版mpv的谜之原生分辨率问题

### DISCUSSION THREAD

**Discussion by verygoodlee** [hooke007/mpv_PlayKit#366] [mpv 通用]
最近更新最新版mpv发现个坑，[builds by shinchiro (git)](https://sourceforge.net/projects/mpv-player-windows/files/64bit-v3/)里面的20230910版本 \
播放某些片源的时候，原生分辨率不太对劲 \
例如[【喵萌Production】★07月新番★[BanG Dream! It’s MyGO!!!!!][13][1080p][简日双语]](https://share.dmhy.org/topics/view/651098_Production_07_BanG_Dream%21_It_s_MyGO%21%21%21%21%21_13_1080p.html)
> 防止不可知原因链接打不开，放个磁力：`magnet:?xt=urn:btih:aff97fcc87f4382317fc8fa1af53da73308125ae`

以下测试 mpv都是纯默认，未添加任何配置文件\
20230910版本出现了奇葩的`1920x1088`原生分辨率，实际输出分辨率应该是根据后面的`[x:0, y:0, w:1920, h:1080]`裁切来的`1920x1080`
![v20230910](https://github.com/hooke007/MPV_lazy/assets/45035465/f39fee8e-eb6f-4a3b-8e08-16a0a155667b)
20230903版本是正常的`1920x1080`
![v20230903](https://github.com/hooke007/MPV_lazy/assets/45035465/bea34661-530b-477b-8faf-135aff1acb9a)

使用`ffmpeg`查看元数据，分辨率是`1920x1080`
![1694832192659](https://github.com/hooke007/MPV_lazy/assets/45035465/43d7aab0-d563-4d9f-a233-4b1d4aa0da32)

实在是看不懂`1920x1088`哪里来的，这个原生分辨率错误，\
会导致在`vs`脚本使用`clip.height`获取的高度错误，\
条件配置写类似`profile-cond=p['height']>1080`的条件也会判断失误

目前我只能先回退旧版本了

**Comment by hooke007** [comment]
https://github.com/mpv-player/mpv/issues/12373

**Reply by verygoodlee** [reply]
加了个新配置`vd-apply-cropping`吗，旧版没这个\
由decoder裁切还是由vo裁切的问题，看说明是vo裁切兼容性更好所以默认关闭的，
但是感觉对升级太不友好了，旧配置文件到新版本表现出不一样的行为

**Reply by hooke007** [reply]
不高强度盯着commit changes根本不知道这个很正常，这么影响大的改变甚至没在interface changes文档里记录，我也不知道说啥了

**Reply by verygoodlee** [reply]
况且这个默认值真的好吗，\
条件配置可以用`p[video-out-params/w]` `p[video-out-params/h]`获取输出分辨率，\
但是`vs`脚本里是只能拿原生分辨率吧，毕竟先经过滤镜再输出，分辨率都不能保证获取正确，问题可大了，\
我昨天就是因为开启补帧脚本发现的这个问题，设置的是`>1080p`的话先缩小到`1080p`再补帧，\
结果这个`1088p`触发缩小了，给我来了个`1904x1080`，左右两条黑边，给我整麻了
![image](https://github.com/hooke007/MPV_lazy/assets/45035465/661b7046-3400-4346-aa28-ed5241202902)

**Reply by hooke007** [reply]
测试启用该选项之后是否工作符合预期

**Reply by verygoodlee** [reply]
启用后正常了，表现与旧版一致

**Reply by zhongfly** [reply]
9.11已经默认启用这个选项了[6f83529](https://github.com/mpv-player/mpv/commit/6f83529f219ab5ac546013967419c00e6ff3edfc)，使得与之前一样。
这么说来，不是“最新版”的问题，是9.8~9.11这个时间段才会有问题
最新版可以用github上发布的版本[shinchiro的每日版本](https://github.com/shinchiro/mpv-winbuild-cmake/releases/latest) 或者 [我的（跟踪mpv git更新）](https://github.com/zhongfly/mpv-winbuild/releases/latest)

**Reply by verygoodlee** [reply]
那就好，最新的确实默认启用了，官方在线文档也可以看到 `Enabled by default.`\
https://mpv.io/manual/master/#options-vd-apply-cropping \
刚好踩到坑里了
